contrib_bots/lib/gmail.py

```diff
@@ -0,0 +1,174 @@
+        label = values['label']
+        subject = values['subject']
+        recipient = values['recipient']
+
+        service = get_service()
+        email = create_message(body, subject, recipient)
+        result = send_message(service, email)
+        print(result)
+
+        label_result = -1
+        if (label != ''):
+            create_label(service, label)
+            label_id = get_label_id(service, label)
+            label_result = add_label(service, label_id, result['id'])
+
+        if ('SENT' in result['labelIds']):
+            if (label_result == 0):
+                client.send_message(dict(
+                    type = 'private',
+                    to = original_sender,
+                    sender = self.bot_email,
+                    content = 'Email sent to {}'.format(recipient)
+                ))
+            else:  # Enters this block if the label couldn't be added or there wasn't a label
+                client.send_message(dict(
+                    type = 'private',
+                    to = original_sender,
+                    sender = self.bot_email,
+                    content = 'Email sent to {} without label'.format(recipient)
+                ))
+        else:
+            client.send_message(dict(
+                type = 'private',
+                to = original_sender,
+                sender = self.bot_email,
+                content = 'Could not send email to {}. Please try again.'.format(recipient)
+            ))
+
+
+def parse_input(user_input):
+    # Input is of the following format:
+    # @gmail --recipient <recipient_email> --subject <subject> --label <label> --body <body>
+    body = user_input.split('--body')[1].strip()
+    content = user_input.split('--body')[0]
+    label = content.split('--label')[1].strip()
+    content = content.split('--label')[0]
+    subject = content.split('--subject')[1].strip()
+    content = content.split('--subject')[0]
+    recipient = content.split('--recipient')[1].strip()
+
+    return {'body': body, 'label': label, 'subject': subject, 'recipient': recipient}
+
+
+def create_message(body, subject, recipient):
+    message = MIMEText(body)
+    message['to'] = recipient
+    message['from'] = 'me'
+    message['subject'] = subject
+
+    raw = base64.urlsafe_b64encode(message.as_bytes())
+    return {'raw': raw.decode()}
+
+
+def send_message(service, email):
+    try:
+        result = service.users().messages().send(userId='me', body=email).execute()
+        return result
+    except errors.HttpError:
+        logging.exception('An error occured while sending the email.')
+        return -1
+
+
+def create_label(service, label):
+    # Creates the label if it doesn't already exist
+    label_object = {'name': label}
+    labels = service.users().labels().list(userId='me').execute()
+    labels = labels['labels']
+
+    label_names = []
+    for l in labels:
+        label_names.append(l['name'])
+
+    if (label not in label_names):
```

Comment:
Hm... I see `label_names` is not needed here.
You can implement it as:
`if label not in [l['name'] for l in labels]`